agent/agents/supervisor/supervisor.go

```diff
@@ -214,6 +244,9 @@ func (s *Supervisor) SetState(state *agentpb.SetStateRequest) {
 }
 
 func (s *Supervisor) storeLastStatus(agentID string, status inventorypb.AgentStatus) {
+	s.agentStatuses.WithLabelValues(agentID).Set(float64(status))
+	s.agentStatusChangeTotal.WithLabelValues(agentID).Inc()
```

Comment:
Is there any reason we update the values before acquiring a lock?

Reply:
Hi, Michal. Fixed a place with mutex. According to linter errors, everything seems to be fixed, maybe with the exception of cyclomatic complexity, but there is just one switch. Regarding the refactoring proposal. No problem. Is it to be done as a service or... ? Maybe in the project there is an example of what you would like to see? I suppose this will also need to be done for the channel?